WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/granular/usecases/PostsAndPagesUseCase.kt

```diff
@@ -36,15 +37,27 @@ constructor(
     private val statsGranularity: StatsGranularity,
     @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher,
     private val postsAndPageViewsStore: PostAndPageViewsStore,
-    private val statsDateFormatter: StatsDateFormatter
+    private val statsDateFormatter: StatsDateFormatter,
+    private val selectedDateProvider: SelectedDateProvider
```

Comment:
I'd consider removing the dependency between UseCases and SelectedDateProvider and just pass the selected date as a parameter to the load and fetch methods. Wdyt?

Reply:
I was thinking about that. The problem is we don't need the date in the `Insights` use cases so we'd either need to create another layer of inheritance (which I don't wanna do) or add a date param to all the use cases (even to those which don't need it) which might be even worse. Let me know if you can think of a cleaner solution